.gitignore

```diff
@@ -102,6 +103,13 @@ node_modules
 testApp
 coverage
 
+############################
+# Meta Tests
+############################
+
+src/api/code-challenge/content-types/code-challenge/*.js
+!src/api/code-challenge/content-types/code-challenge/lifecycles.js
```

Comment:
This is my "scalable" way of dealing with the .js files created during typescript compilation. 

It was a long time ago, but I believe lifecycles.js had an issue with typescript since pMap wasn't compatible with the import ES6 syntax, we used require.

We originally discussed it here: 
https://github.com/oriooctopus/acc-v4/pull/8#discussion_r881661141

Reply:
I would love it if you did that.

Reply:
Also, I don't think the compiled files should go in the same folder as the uncompiled files. I think they should go in a`dist` folder or something. That would automatically solve the gitignore issue as well

Reply:
> I would love it if you did that.

``` To test the beta version of TypeScript support, run the command npx create-strapi-app@beta my-app (--ts) in a terminal. ```
I will need to rebuild the app. I will save this for the last part.

Can I restore the previous version using git reset --hard ?

Reply:
I've decided to wait a month for strapi to continue working out bugs with their beta typescript support

Reply:
> Also, I don't think the compiled files should go in the same folder as the uncompiled files. I think they should go in a`dist` folder or something. That would automatically solve the gitignore issue as well

I agree with the dist folder, but should the dist folder be a clone of the src folder or just a smaller folder within src? Currently very few files are being used with TS, and TS support is still in beta for strapi

Reply:
I will wait until strapi integrates TS to implement the outDir & srcDir, since I'm not sure if running "strapi start" / "strapi develop" will work with a dist folder yet.